apps/hash-frontend/src/pages/[shortname].page/edit-user-profile-info-modal/user-profile-info-form.tsx

```diff
@@ -0,0 +1,359 @@
+    ],
+  );
+
+  const innerSubmit = handleSubmit(async (data) => {
+    setLoading(true);
+
+    const { serviceAccounts, ...userData } = data;
+
+    await Promise.all([
+      updateServiceAccounts({ serviceAccounts }),
+      updateUser(userData),
+    ]);
+
+    /** @todo: error handling */
+
+    setLoading(false);
+
+    void refetchUserProfile();
+
+    closeModal();
+  });
+
+  const handleDiscard = useCallback(() => {
+    reset();
+    closeModal();
+  }, [reset, closeModal]);
+
+  const isSubmitDisabled =
+    Object.keys(errors).length > 0 || Object.keys(dirtyFields).length === 0;
+
+  return (
+    <Box
+      component="form"
+      onSubmit={innerSubmit}
+      sx={{
+        marginTop: 2,
+        "> :not(:last-child)": {
+          marginBottom: 3,
+        },
+      }}
+    >
+      <FormProvider {...formMethods}>
+        <TextField
+          id="name"
+          fullWidth
+          label="Preferred name"
+          placeholder="Enter your preferred name"
+          required
+          error={touchedFields.preferredName && !!errors.preferredName}
+          {...register("preferredName", { required: true })}
+        />
+        <TextField
+          fullWidth
+          label="Location"
+          placeholder="Enter your current city/location"
+          {...register("location")}
+        />
+        <TextField
+          fullWidth
+          label="Website URL"
+          placeholder="Enter a website, e.g. https://example.com/"
+          error={touchedFields.website && !!errors.website}
+          {...register("website", {
+            pattern: {
+              value: urlRegex,
+              message: "Please enter a valid URL",
+            },
+          })}
+        />
+        <RHFSelect
+          control={control}
+          name="preferredPronouns"
+          defaultValue=""
+          selectProps={{
+            label: "Preferred pronouns",
+            displayEmpty: true,
+            placeholder: "Select pronouns (he/him, she/her, they/them)",
+            renderValue: (selected) =>
+              selected === "" ? (
+                <Box
+                  component="span"
+                  sx={{
+                    color: ({ palette }) => palette.gray[50],
+                  }}
+                >
+                  Select pronouns (he/him, she/her, they/them)
+                </Box>
+              ) : (
+                <>{String(selected)}</>
+              ),
+          }}
+        >
+          <MenuItem value="">None</MenuItem>
+          <MenuItem value="he/him">he/him</MenuItem>
+          <MenuItem value="she/her">she/her</MenuItem>
+          <MenuItem value="they/them">they/them</MenuItem>
+        </RHFSelect>
+        <ServiceAccountsInput />
+        <Box display="flex" columnGap={1.5}>
+          <Button type="submit" loading={loading} disabled={isSubmitDisabled}>
+            Save changes
+          </Button>
+          <Button variant="tertiary" onClick={handleDiscard}>
+            Discard
```

Comment:
```suggestion
            {Object.keys(dirtyFields).length === 0 ? "Cancel" : "Discard"}
```

Reply:
Addressed in https://github.com/hashintel/hash/pull/3338/commits/12d4bb0bf97b6a46dadee2bff9095106161e2b45